include/dgl/aten/macro.h

```diff
@@ -152,8 +152,13 @@
         XPU == kDGLCUDA && (val).bits == 16 && (val).code == kDGLBfloat) {  \
       typedef __nv_bfloat16 FloatType;                                      \
       { __VA_ARGS__ }                                                       \
-    } else if (XPU == kDGLCPU) {                                            \
-      LOG(FATAL) << (val_name) << " can only be float32 or float64 on CPU"; \
+    } else if (                                                             \
+        XPU == kDGLCPU && (val).bits == 16 && (val).code == kDGLFloat) {    \
+      LOG(FATAL) << (val_name) << " can't be float16 on CPU";               \
+    } else if (                                                             \
+        XPU == kDGLCPU && (val).bits == 16 && (val).code == kDGLBfloat) {   \
+        typedef BFloat16 FloatType;                                         \
+      { __VA_ARGS__ }                                                       \
```

Comment:
Can you run lintrunner -a?
I think there are some issue about different new line characters, which we can't see from eyes about lint catches it.

Reply:
changed